### PR TITLE
workflows/tests: skip tap syntax jobs if `CI-published-bottle-commits`

### PR DESCRIPTION
**Pull request by ZhongRuoyu**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

This should allow maintainers to merge PRs as soon as bottle commits are published. Tap syntax is always checked on pushes to `master`, so I see some redundancy here.

### CODE REVIEW

**Comment by ZhongRuoyu** [comment]
Without https://github.com/Homebrew/homebrew-core/labels/CI-published-bottle-commits: https://github.com/Homebrew/homebrew-core/actions/runs/4525850696/jobs/7970621011?pr=126733

With https://github.com/Homebrew/homebrew-core/labels/CI-published-bottle-commits: https://github.com/Homebrew/homebrew-core/actions/runs/4525956739/jobs/7970801339?pr=126733

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -37,13 +37,15 @@ jobs:
       deleted_formulae: ${{ steps.formulae-detect.outputs.deleted_formulae }}
     steps:
       - name: Set up Homebrew
+        if: github.event_name != 'pull_request' || !contains(github.event.pull_request.labels.*.name, 'CI-published-bottle-commits')
```
Comment (carlocab) [inline .github/workflows/tests.yml:40]: Should we move these three statements up here?

https://github.com/Homebrew/homebrew-core/blob/a236345fdfbb37465edc0b5a0ad1a51dcbd84423/.github/workflows/tests.yml#L28

Comment (ZhongRuoyu) [reply .github/workflows/tests.yml:40]: Ah, yes, I think we should. In that way `setup_tests` (and subsequently, `tests`) may also be skipped due to `needs: tap_syntax`. But that's fine.

https://github.com/Homebrew/homebrew-core/blob/1828a10760f818891bd199ca8a057c330354d558/.github/workflows/tests.yml#L51-L56

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -25,7 +25,7 @@ permissions:
 
 jobs:
   tap_syntax:
-    if: github.repository == 'Homebrew/homebrew-core'
+    if: github.repository == 'Homebrew/homebrew-core' && (github.event_name != 'pull_request' || !contains(github.event.pull_request.labels.*.name, 'CI-published-bottle-commits'))
```
Comment (ZhongRuoyu) [inline .github/workflows/tests.yml:28]: I hear De Morgan talking to me.

```suggestion
    if: github.repository == 'Homebrew/homebrew-core' && !(github.event_name == 'pull_request' && contains(github.event.pull_request.labels.*.name, 'CI-published-bottle-commits'))
```

**Review by carlocab — APPROVED** [review]
Ok with this. It does mean that the bottle commit doesn't get the tap syntax test until it gets committed to `master`, but that's the case already with the existing merge flow.

Of course, this would prevent us from making the tap syntax test a mandatory status check for merging, but I'm not sure that's needed. Tap syntax failures will break any PR opened after they're merged.

**Comment by p-linnane** [comment]
Had no issues merging https://github.com/Homebrew/homebrew-core/pull/126735 utilizing this.

**Comment by MikeMcQuaid** [comment]
Sorry, strongly 👎🏻 on this, would appreciate it being reverted please @ZhongRuoyu or @carlocab 

> It does mean that the bottle commit doesn't get the tap syntax test until it gets committed to `master`, but that's the case already with the existing merge flow.

This is bad, though. It means that a bad bottle commit can break CI on `master` (this has happened multiple times in the past). It also means that you can effectively use this flow to push commits to `master` without any testing and that we cannot ever make even the syntax job a required build.

If anything: we should be adding _more_ tests that are run before merging (i.e. running `brew fetch` to verify the bottles were actually published as expected) not fewer.

> Of course, this would prevent us from making the tap syntax test a mandatory status check for merging, but I'm not sure that's needed.

It actually prevents us from making _any test mandatory for merging_ which is pretty much the whole point of pushing bottle commits to PRs, to get away from the current flow where we're pushing commits to `master` without any verification that they actually work 😢.

**Comment by MikeMcQuaid** [comment]
Opened https://github.com/Homebrew/homebrew-core/pull/126790 to revert

**Pull request merged**